Core.ApplicationServices/SystemUsage/Migration/ItSystemUsageMigrationService.cs

```diff
@@ -36,87 +36,107 @@ public class ItSystemUsageMigrationService : IItSystemUsageMigrationService
             ILogger logger,
             IItSystemRepository systemRepository,
             IItSystemUsageRepository systemUsageRepository,
-            IItsystemUsageRelationsService itsystemUsageRelationsService,
+            IItsystemUsageRelationsService itSystemUsageRelationsService,
             IDomainEvents domainEvents)
         {
             _authorizationContext = authorizationContext;
             _transactionManager = transactionManager;
             _logger = logger;
             _systemRepository = systemRepository;
             _systemUsageRepository = systemUsageRepository;
-            _itsystemUsageRelationsService = itsystemUsageRelationsService;
+            _itSystemUsageRelationsService = itSystemUsageRelationsService;
             _domainEvents = domainEvents;
         }
 
-        public Result<IReadOnlyList<ItSystem>, OperationFailure> GetUnusedItSystemsByOrganization(
+        public Result<IQueryable<ItSystem>, OperationError> GetUnusedItSystemsByOrganizationQuery(
             int organizationId,
-            string nameContent,
             int numberOfItSystems,
-            bool getPublicFromOtherOrganizations)
+            bool getPublicFromOtherOrganizations,
+            params IDomainQuery<ItSystem>[] conditions)
         {
-            if (string.IsNullOrWhiteSpace(nameContent))
+            if (numberOfItSystems is < 1 or > 25)
```

Comment:
this rule was present in the v1 Controller, should I ditch it, or leave it like that?

Reply:
it's ok to have it here